### DISCUSSION TITLE
Understanding the codebase

### DISCUSSION THREAD

**Discussion by Vadi** [jscad/OpenJSCAD.org#1351] [Show and tell]
I came across and blown away how great the codebase looks. Thanks for the great effort.

I'm trying to understand the codebase particularly from which point geometry (from modelling) is generated to triangles / lines to pass onto webgl for rendering. Can someone point me to a file/code/function.

Thanks in advance.

**Comment by hrgdavor** [comment]
start here: https://github.com/jscad/OpenJSCAD.org/blob/master/packages/utils/regl-renderer/src/geometry-utils-V2/entitiesFromSolids.js

**Reply by Vadi** [reply]
I figured demo.html that @z3dev  pointed uses `entitiesFromSolids`. Thanks.

**Comment by z3dev** [comment]
@Vadi Take a look at the file, demo.html. This is located in the utils/regl-renderer directory. It will provide a simple JSCAD design, which is rendered via REGL. From that example, you should be able to understand the logic in the rendering, converting JSCAD geometry to REGL buffers, and rendering.

**Reply by Vadi** [reply]
This gives me a great start.
